project/frontend/src/App.js

```diff
@@ -36,57 +41,57 @@ function App() {
     setCompId(parsedCompId);
   }, []);
 
-  if (user.signedIn && !user.id) {
-    //TODO: fetch for id... with useremail param... once backend is ready
-    //for now id hardcoded to 123
-    setUser((prevState) => {
+  const fetchCompetitionInfo = async (fetchCall, stateKey) => {
+    try {
+      const data = await fetch(fetchCall).then((response) => response.json());
+
       return {
-        ...prevState,
-        id: 1,
+        [stateKey]: data,
       };
-    });
-  }
-
-  const fetchAndUpdateCompetitionInfo = async (fetchCall, stateKey) => {
-    return fetch(fetchCall)
-      .then((response) => response.json())
-      .then((data) => {
-        setCompetitionInfo((prevState) => {
-          return {
-            ...prevState,
-            [stateKey]: data,
-          };
-        });
-      });
+    } catch (e) {
+      return {
+        [stateKey]: undefined,
+      };
+    }
   };
 
   React.useEffect(() => {
+    if (!authStateReceived) {
+      return;
+    }
+
     if (user.signedIn && user.id && compId) {
```

Comment:
While we're refactoring - I think this logic should be moved into a named function to better express what we're testing for here.
Also I think we should be more explicit about what is falsy - coercing integers into booleans can lead to some annoying errors (in general, not specifically here).

Reply:
Done :D